### PR TITLE
Render PlantUML diagrams in HTML export

### PR DESCRIPTION
**Pull request by tangoalx**
According to #335.

By default: accessing http://www.plantuml.com/plantuml to generate plantuml diagrams.

Images are integrated in HTML instead of stored in asset folder.

REQ002.yml extended to show exemplary usage in tests. Updated tests accordingly.

### CODE REVIEW

File: doorstop/core/publisher.py [hunk 1]
```diff
@@ -18,7 +19,13 @@
     'markdown.extensions.extra',
     'markdown.extensions.sane_lists',
     'mdx_outline',
-    'mdx_math'
+    'mdx_math',
+    PlantUMLMarkdownExtension(server='http://www.plantuml.com/plantuml',
+                              cachedir='/tmp',
```
Comment (jacebrowning) [inline doorstop/core/publisher.py]: Will this work on Windows? Perhaps `tempfile.gettempdir()` will be more portable.

Comment (tangoalx) [reply doorstop/core/publisher.py]: Unfortunately, I don't have a windows system available to test it. I can change it anyway to see if it still works.

**Comment by jacebrowning** [comment]
Thanks for putting this together! Please add an example diagram in [`TUT017`](https://github.com/jacebrowning/doorstop/blob/develop/reqs/tutorial/TUT017.yml) (or make a new item) to show off this feature.

**Comment by tangoalx** [comment]
@jacebrowning: how is the tutorial updated on github.io? Do you do it if a new release is published?

**Comment by jacebrowning** [comment]
@tangoalx This was supposed to be happening automatically, but it looks like my token expired.

Your changes are now live: http://jacebrowning.github.io/doorstop/TUT.html#TUT017

**Pull request merged**